common/lib/xmodule/xmodule/modulestore/tests/django_utils.py

```diff
@@ -317,10 +317,15 @@ def test_that_edits_modulestore(self):
         @functools.wraps(f)
         def wrapper(*args, **kwargs):
             """Call the object method, and reset the test case afterwards."""
-            return_val = f(*args, **kwargs)
-            obj = args[0]
-            obj.reset()
-            return return_val
+            try:
+                return_val = f(*args, **kwargs)
+            except:
+                raise
+            else:
+                return return_val
+            finally:
+                obj = args[0]
+                obj.reset()
```

Comment:
This code looks correct to me, but may also confuse a lot of people. Few people use the `else` clause with `try`, or are familiar with what happens when you return before a `finally` block. Could you please add comments explaining what's going on?